build.gradle

```diff
@@ -227,6 +234,28 @@ subprojects {
                 compileClasspath += generated.output
                 runtimeClasspath += generated.output
             }
+            scala {
+                compileClasspath += generated.output
+                runtimeClasspath += generated.output
+            }
+            scoverage {
+                compileClasspath += generated.output
+                runtimeClasspath += generated.output
+            }
+            testScoverage {
+                compileClasspath += generated.output
+                runtimeClasspath += generated.output
+            }
+        }
+        test {
+            java {
```

Comment:
These super shouldn't be necessary, because it takes all the main classpaths and adds them to test by default?

Reply:
It does, which is why our tests were running prior to this change. However, adding this enabled IntelliJ to recognize generates sources in test code, which it has not been doing for me prior.

It would seem that generated classes are added to same output directory as Java classes, which is why they totally work in tests (i.e., are on the runtime classpath). However, IntelliJ does not know about the sources, so it highlights things in red.

For a little more context, I added some printlns so that the configuration looks like this:
```
        test {
            java {
                println("java compileClasspath: " + compileClasspath.files)
                println("java runtimeClasspath: " + runtimeClasspath.files)
                compileClasspath += generated.output
                runtimeClasspath += generated.output
                println("java compileClasspath: " + compileClasspath.files)
                println("java runtimeClasspath: " + runtimeClasspath.files)
            }
            ...
        }
```
and here is the output:
```
java compileClasspath: [/Users/dami6356/dev/repose/repose-aggregator/build/classes/main, /Users/dami6356/dev/repose/repose-aggregator/build/resources/main]
java runtimeClasspath: [/Users/dami6356/dev/repose/repose-aggregator/build/classes/test, /Users/dami6356/dev/repose/repose-aggregator/build/resources/test, /Users/dami6356/dev/repose/repose-aggregator/build/classes/main, /Users/dami6356/dev/repose/repose-aggregator/build/resources/main]
java compileClasspath: [/Users/dami6356/dev/repose/repose-aggregator/build/classes/main, /Users/dami6356/dev/repose/repose-aggregator/build/resources/main, /Users/dami6356/dev/repose/repose-aggregator/build/resources/generated]
java runtimeClasspath: [/Users/dami6356/dev/repose/repose-aggregator/build/classes/test, /Users/dami6356/dev/repose/repose-aggregator/build/resources/test, /Users/dami6356/dev/repose/repose-aggregator/build/classes/main, /Users/dami6356/dev/repose/repose-aggregator/build/resources/main, /Users/dami6356/dev/repose/repose-aggregator/build/resources/generated]
```

So as you can see, the generated resources directories are not on the classpath without this change... but we don't really care about those.

Which makes me wonder why this makes IntelliJ happy. They way IntelliJ utilizes Gradle must be odd, as adding directories we don't really care about to the `compileClasspath` seems to be helping IntelliJ find the generates sources. Instead, I think I should probably just add the `generatedDir` to the `srcDirs` for the configurations in the `test` source set. Does that make sense? I need to test it and see what happens, but that feels more right.

Reply:
Quick update -- I just testing setting the `srcDirs` and that also seems to satisfy IntelliJ. Plus, really, that's what I wanted to do in the first place.

Reply:
Hah. An unfortunate, but amusing, side-effect is that checkstyle will look at generated sources, which obviously fail the check.

Reply:
So `generated.output` is a `SourceSetOutput` that just holds references to a classes output directory and a resources output directory. Since the classes output directory is set to the main classes output directory, the practical effect of all of the lines doing something like `compileClasspath += generated.output` is to add the generated resources directory to the classpath. But the generated resources directory is always empty.

So then why did we add `compileClasspath += generated.output` and the like? Presumably to make IntelliJ happy, since we've established that generated classes live in the same directory as compiled code classes. And why does adding that make IntelliJ happy? That's the thing I can't answer.

Reply:
Man i super don't remember, but ti did take a lot of exploration and poking at things to get intellij a little happy.

Reply:
So we can remove the Scala `SourceDirectorySet` configuration since it extends the Java one anyway (as described in the comments on the other line), but I think we should keep the Java one. It certainly doesn't harm anything, and give us (or at least me) a nicer experience in IntelliJ.